### ISSUE TITLE
Add `cancel()` to `RunningTask` and deprecate `kill()`

### ISSUE THREAD

**Issue by marlonjames** [cocotb/cocotb#2679]
Continuing the theme of providing the new interfaces before 2.0.
For now we could keep it the same, without using `CancelledError`.

Potentially, `kill()` could be made private and used by the scheduler, whereby the task is cancelled even if it tries to suppress the exception.

**Comment by ktbarrett** [comment]
So the point is just the name change? I think if we are going to introduce "another way" to kill a task it should behave at least slightly differently.

**Comment by marlonjames** [comment]
Yes the point is to change the name and refer to cancelling a task. Brings the language in line with `asyncio` and `trio`.

**Comment by ktbarrett** [comment]
Are there problems with making `cancel` throw `CancelledError` right away? Is that a lot of work? If it is, perhaps we should emit a `FutureWarning` about it throwing `CancelledError` in 2.0?

**Comment by marlonjames** [comment]
It would take some work, as `kill` just calls `_unschedule`, which doesn't run the `Task`'s coroutine again.

**Comment by marlonjames** [comment]
`cancel()` was added in #2877.
Deprecating `kill()` is to-be-decided.

**Comment by ktbarrett** [comment]
I think kill() should be deprecated and I think we should also work to support throwing CancelledErrors.
